rmw_fastrtps_cpp/src/type_support_common.cpp

```diff
@@ -129,14 +163,95 @@ bool TypeSupport::deserializeROSmessage(
   return true;
 }
 
-MessageTypeSupport::MessageTypeSupport(const message_type_support_callbacks_t * members)
+bool TypeSupport::getKeyHashFromROSmessage(
+    void * ros_message,
+    eprosima::fastrtps::rtps::InstanceHandle_t * ihandle,
+    bool force_md5,
+    const void * impl) const
+{
+  assert(ros_message);
+  (void)impl;
+
+  /*if (capacity < 16)
+  {
+    throw std::runtime_error("Not enough capacity to serialize key");
+  }*/
+
+  //! retrieve estimated serialized size in case key is unbounded
+  if (this->key_is_unbounded_)
+  {
+    this->key_max_serialized_size_ = key_callbacks_->get_serialized_key_size(ros_message, 0);
+    this->key_buffer_.reserve(this->key_max_serialized_size_);
+  }
```

Comment:
```suggestion
  // retrieve estimated serialized size in case key is unbounded
  if (key_is_unbounded_)
  {
    key_max_serialized_size_ = (std::max)((key_max_serialized_size_, key_callbacks_->get_serialized_key_size(ros_message, 0));
    key_buffer_.reserve(key_max_serialized_size_);
  }
```